pkg/receiver/mysqlrecordsreceiver/client.go

```diff
@@ -0,0 +1,343 @@
+//Details : https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL.html
+func createIAMRDSTLSConf(pempath string, logger *zap.Logger) tls.Config {
+	rootCertPool := x509.NewCertPool()
+	globalpem, err := ioutil.ReadFile(pempath)
+	if err != nil {
+		logger.Error("error in reading pem file", zap.Error(err))
+	}
+	if ok := rootCertPool.AppendCertsFromPEM(globalpem); !ok {
+		logger.Error("error in loading certificates from pem file", zap.Error(err))
+	}
+	return tls.Config{
+		RootCAs: rootCertPool,
+	}
+}
+
+//This function calls for the AWS packaged API which will generate an authentication token that can be used for accessing a AWS RDS instance instead of a password.
+//Details : https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.IAMDBAuth.html
+func generateIAMAuthToken(endpoint string, conf *Config, logger *zap.Logger) (token string) {
+	cfg, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		logger.Error("configuration error:", zap.Error(err))
+	}
+	authenticationToken, err := auth.BuildAuthToken(
+		context.TODO(), endpoint, conf.Region, conf.Username, cfg.Credentials)
+	if err != nil {
+		logger.Error("failed to create authentication token:", zap.Error(err))
+	}
+	return authenticationToken
+}
+
+//There are 3 scenarios here for creating connection strings for a database connection
+//1. With a plaintext password
+//2. With an encrypted plaintext password
+//3. With an AWS Authentication token to be used as a password
+func newMySQLClient(conf *Config, logger *zap.Logger) client {
+	var port string
+	var basicauthpassword string
+	var connStr string
+	basicauthpassword = conf.Password
+	//Encrypting a plaintext password if a 24 character secret string is provided by the user from an external file
+	if (len(conf.PasswordType) == 0 || conf.PasswordType == "plaintext") && len(conf.EncryptSecretPath) != 0 {
+		MySecret1, err := readMySecret(conf)
+		if err != nil {
+			logger.Error("error in reading encryption secret from file", zap.Error(err))
+		}
+		encText, err := Encrypt(conf.Password, MySecret1, logger)
+		if err != nil {
+			logger.Error("error encrypting your classified text", zap.Error(err))
+		}
+		logger.Debug("The plaintext password can be replaced with this encrypted password.", zap.String("encryptedPassword", encText))
+	}
+	//Decrypting an encrypted password
+	if conf.PasswordType == "encrypted" {
+		MySecret2, err := readMySecret(conf)
+		if err != nil {
+			logger.Error("error in reading encryption secret from file", zap.Error(err))
+		}
+		decText, err := Decrypt(conf.Password, MySecret2, logger)
+		if err != nil {
+			logger.Error("error decrypting your encrypted text: ", zap.Error(err))
+		}
+		basicauthpassword = decText
+	}
+	//Considering default port to be 3306 if not provided by user
+	if len(conf.DBPort) != 0 {
+		port = conf.DBPort
+	} else {
+		logger.Info("dbport empty, considering default 3306")
+		port = "3306"
+	}
+	endpoint := conf.DBHost + ":" + port
```

Comment:
I there a need to specify port and host separately?

Reply:
not really but the separation gives us more granularity.

Reply:
Do we support connection strings?
e.g. `mysql://user@localhost:3306`

Reply:
nope, we build the connection string, that's how it generally is.

Reply:
Why do we do that?

I personally prefer to have connection string in configuration
I believe it is easier for developer as well as for user

Reply:
I believe building the conn str gives us a larger window for feature implementation as developers, saw examples in our present python db integrator as well as the open contrib collector.